### ISSUE TITLE
AWS STS Documentation incorrect. Suggests Credentials meet provideCredentials crate

### ISSUE THREAD

**Issue by spf5000** [awslabs/aws-sdk-rust#737]
### Describe the bug

Currently on the aws-sdk-sts documentation for the [Credentials](https://docs.rs/aws-sdk-sts/latest/aws_sdk_sts/struct.Credentials.html) struct, it shows an [impl](https://docs.rs/aws-sdk-sts/latest/aws_sdk_sts/struct.Credentials.html#impl-ProvideCredentials-for-Credentials) for the aws_credentials_types [ProvideCredentials](https://docs.rs/aws-credential-types/0.54.1/aws_credential_types/provider/trait.ProvideCredentials.html) trait. However it appears that the modeled version from STS does not implement this trait and is referencing the [Credentials struct in aws_credentials_types](https://docs.rs/aws-credential-types/0.54.1/aws_credential_types/struct.Credentials.html). 

### Expected Behavior

I'd expect the STS modeled credentials to either implement ProvideCredentials or not document this as a trait implementation.

### Current Behavior

The docs show the ProvideCredentials trait is implemented, but it can't compile

### Reproduction Steps

```
async fn get_credentials_helper() -> Box<dyn ProvideCredentials {
    let shared_config = aws_config::load_from_env().await;
    let client = Client::new(&shared_config);
    let req = client
        .assume_role()
        .role_arn("myAwesomeRole")
        .role_session_name("Test");
    let creds: &StsCreds = req.send()
        .await.unwrap()
        .credentials()
        .unwrap();

    Box::new(creds)
}
```

### Version

```text
aws_credentials_types = v0.54.1
aws_sdk_sts = v0.24.0
```

### Environment details (OS name and version, etc.)

Amazon Linux 2 aarch64

**Comment by Velfi** [comment]
Hey @spf5000, thanks for submitting this issue. We'll add it to our backlog.

**Comment by ysaito1001** [comment]
Hi, `aws_sdk_sts::Credentials` is a [re-export](https://github.com/awslabs/aws-sdk-rust/blob/66423e05991ee831696bc32fe3e452694cf0d231/sdk/sts/src/lib.rs#L88) of `aws_credential_types::Credentials`. Since the latter [implements](https://github.com/awslabs/aws-sdk-rust/blob/66423e05991ee831696bc32fe3e452694cf0d231/sdk/aws-credential-types/src/provider.rs#L285-L292) the `ProvideCredentials` trait, the former implements that trait automatically.

We've been able to compile the code in the reproduction steps with `aws_credentials_types = 0.54.1` and
`aws_sdk_sts = 0.24.0`.

Could you provide us with the compiler error message you are seeing so that we can better understand what could be the issue on your end?

**Comment by rcoh** [comment]
I'd add as a follow up—rather than using the STS credentials directly, I'd recommend using the credentials providers in `aws-config` as they implement best practices around caching, expiry, and secrets handling.

**Comment by spf5000** [comment]
Sorry for the slow replies here. Below is a full example reproducing this issue:

src/main.rs:

```
use aws_credential_types::provider::ProvideCredentials;
use aws_sdk_sts::model::Credentials;
use aws_sdk_sts::Client;

fn main() {
    println!("Hello, world!");

    get_credentials_helper();
}

#[tokio::main]
async fn get_credentials_helper() -> Box<dyn ProvideCredentials> {
    let shared_config = aws_config::load_from_env().await;
    let client = Client::new(&shared_config);
    let req = client
        .assume_role()
        .role_arn("myAwesomeRole")
        .role_session_name("Test");
    let creds: &Credentials = req.send()
        .await.unwrap()
        .credentials()
        .unwrap();

    Box::new(creds.clone())
}
```

Cargo.toml:

```
[package]
name = "rust-sdk-bug"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
aws-credential-types = "0.54.1"
aws-sdk-sts = "0.24.0"
tokio = { version = "1", features = ["full"] }
aws-config = "0.54.1"
```

**Comment by spf5000** [comment]
To be clear, I understand that I can leverage the `Credentials` struct in `aws_credential_types`, but I'm trying to implement a simple function that will return a `Box<dyn ProvideCredentials>` that way I can update this function with different credential providers if/as needed without callers of the function being tied to a specific `ProvideCredentials` implementation. I know I can copy over the data from the `aws_sdk_sts::model::Credentials` into a `aws_credentias_types::Credentials` struct, but the documentation (and re-export) suggests that I shouldn't have to.

**Comment by spf5000** [comment]
I've also tried updating the `aws_sdk_sts::model::Credentials` to `use aws_credential_types::Credentials` and I've tried removing the export and the explicit type assignment in `let creds: &Credentials = req.send()` (ex: `let creds: = req.send()`, but I'm still seeing the same compiler error. I'm using the 1.67 rust compiler (rustc) as well if that helps.

**Comment by spf5000** [comment]
Makes sense. Any plans to clarify that in the documentation? Adding a link from the STS crate might help avoid future confusion from folks.

**Comment by rcoh** [comment]
- [ ] clarify documentation on credentials re-export in all client crates
